### ISSUE TITLE
Sony DSC-WX800 Support

### ISSUE THREAD

**Issue by ijcd** [ma1co/Sony-PMCA-RE#293]
I have one of these cameras and am looking into getting support added for it. Before I go too far, I saw this:

"A list of supported camera models can be found [here](https://github.com/ma1co/fwtool.py/blob/master/devices.yml). gen4 models are not compatible, since their firmware is cryptographically signed."

What are the gen4 models? I don't want to embark on a fool's errand if this will not work.

Otherswise, I am working through the steps here: https://github.com/ma1co/Sony-PMCA-RE/issues/196

**Comment by ijcd** [comment]
Working on this, but seems the driver might not be being found:

```
[nix-shell:~/work/sony-camera]$ ./Sony-PMCA-RE/pmca-console.py guess_firmware -f 0x51020000.zip
Using drivers OS-X-MSC, libusb-MTP
Looking for Sony devices
No devices found. Ensure your camera is connected.
```

**Comment by ijcd** [comment]
I'm now trying to figure out how to get the pmca-console updatershell to start and recognize the camera. Currently:

```
[nix-shell:~/work/sony-camera]$ ./Sony-PMCA-RE/pmca-console.py updatershell
Using drivers OS-X-MSC, libusb-MTP
Looking for Sony devices

Querying mass storage device
Sony DSC is a camera in mass storage mode

Getting device info
Using firmware for model DSC-WX800

Unknown device: DSC-WX800
```

I believe, somehow, that the `fwtool` needs to be used to pull out the `.hdr` and `.dat` files so the process described [here](https://github.com/ma1co/Sony-PMCA-RE/blob/2cdf1263a84b62fe440a933ad4c3af915dfd42df/docs/AppInstallation.md) can be used. I don't see any updates for this camera, yet, so presumably I need to figure out how to run a `dd` on the camera? Perhaps using one of the other shells? (`serviceshell`?)

Basically, how do I bootstrap this process for a new camera with no `exe` updates yet?

**Comment by ijcd** [comment]
I did try the `guess_firmware` process with each of these, which I'd found through the issues in this repo:

```
0x11020000.zip
0x41020000.zip
0x51020000.zip
0x61020000.zip
```

None of those report a match.

**Comment by ijcd** [comment]
I think I'm at an impasse for now, hopefully someone will respond with next steps I can attempt. I tried using `servicemode` but the camera drops out of USB mass storage mode and just goes back to photo mode (out of settings). The OSX driver seems to unmount everything. Feels like a crash... or maybe the camera is not in Senser Mode. I can't find any information on how to put it into that mode.

Some docs on how to create the `guess_firmware`zips or how to bootstrap a new camera would go a long way. Many of the issues seem to be about new camera additions.

```
[nix-shell:~/work/sony-camera]$ sudo ./Sony-PMCA-RE/pmca-console.py serviceshell
Password:
Using drivers libusb-MSC, libusb-MTP, libusb-vendor-specific
Looking for Sony devices

Querying mass storage device
Sony DSC is a camera in mass storage mode

Switching to senser mode

Waiting for camera to switch...
Found a camera in senser mode
Operation timed out. Please run this command again when your camera has connected.
```

**Comment by ijcd** [comment]
Would really love some help here. I'm willing to do the work on this end.

**Comment by ijcd** [comment]
Thanks for the response -- I'll give senser mode a try using the updated code and report back. My ultimate goal is to get this camera switched to English -- will that be possible using senser mode if the new work is added? If some direction is provided, I can attempt that work.

**Comment by ijcd** [comment]
If this architecture isn't supported, what does this line do:

https://github.com/ma1co/fwtool.py/blob/12a7c85e8a221524133621a7948a2f6facf3ab0b/fwtool/sony/fdat.py#L223

**Comment by ma1co** [comment]
[Decryption only.](https://github.com/ma1co/fwtool.py/blob/12a7c85e8a221524133621a7948a2f6facf3ab0b/fwtool/sony/fdat.py#L212)

Btw, tweaks should now be supported in serviceshell ([driver installation required](https://github.com/ma1co/Sony-PMCA-RE#windows-drivers)). If you manage to run it, I would be interested in a firmware dump. You can dump the firmware using `pull /dev/nflashaB0` and `pull /dev/nflasha` in serviceshell. Please upload the files to my [dropbox](https://www.dropbox.com/request/WXQ1IKIoPo4ECK8CFxRx). Thanks!
